lib/Serialization/Deserialization.cpp

```diff
@@ -1226,6 +1227,8 @@ static void filterValues(Type expectedTy, ModuleDecl *expectedModule,
       return true;
     if (value->isStatic() != isStatic)
       return true;
+    if (value->hasClangNode() != importedFromClang)
```

Comment:
IIRC, `hasClangNode`() won't be true for `NS_ERROR_ENUM` struct types, so you likely have the same issue with cross-references to those.

Reply:
Yeah, I was going to punt on that. I suppose this could all be a `getModuleScopeContext` check instead.